akamai/util.go

```diff
@@ -97,3 +103,45 @@ func makeCache() *cache.Cache {
 	profilecache := cache.New(5*time.Minute, 10*time.Minute)
 	return profilecache
 }
+
+func setCorrelationID(correlationid string) {
+	//edge.SetupLogging()
+	//log.Printf("[DEBUG] setCorrelationID %s\n", correlationid)
+	edge.SetLogCorrelationId(correlationid)
```

Comment:
Won't concurrent calls to this by parallel resources overwrite each other's values in edge?

Reply:
I tested and because a shared variable is used the end tags are wrong: (trimmed example to shore the core problem)

START CORRELATION ID 305
Reading CP Code
START CORRELATION ID dd0
Reading CP Code
START CORRELATION ID 103
Reading CP Code
data.akamai_cp_code.example3:cpc_309358
END CORRELATION ID 103
data.akamai_cp_code.example2:cpc_23506
END CORRELATION ID 103

(third end statement was missing because shared variable had already been cleared)

Reply:
Each resource apply another thread so if we can get variable to be per instance of call should hold up better (was just trying to get that start end mark to work easier to mark lines I would think)

Reply:
Making it per instance would make it work better but the interleaved calls mean that inline guid is still more efficient way to track a failed call for example.